lib/models/organisation.js

```diff
@@ -217,98 +217,108 @@ schema.post('init', function handleInit() {
 schema.pre('save', async function preSave(next) {
   const User = getConnection().model('User');
   const Role = getConnection().model('Role');
+  const organisation = this;
+  const objectId = mongoose.Types.ObjectId;
 
   // https://github.com/Automattic/mongoose/issues/1474
   this.wasNew = this.isNew;
 
-  const organisation = this;
-  const objectId = mongoose.Types.ObjectId;
-
   if (!isUndefined(this._oldExpiration) && this.expiration !== this._oldExpiration) {
     this.expirationNotifications.expirationNotificationSent = EMAIL_NOOP;
     this.expirationNotifications.weekBeforeNotificationSent = EMAIL_NOOP;
   }
 
+  if (!isUndefined(this._oldTimezone) && this.timezone !== this._oldTimezone) {
+    await update$dteTimezoneInDB(organisation._id, this.timezone);
+  }
+
+  await User.updateMany(
+    { ownerOrganisation: organisation._id },
+    { ownerOrganisationSettings: this.settings },
+    { runValidators: false },
+  );
+
   // https://github.com/Automattic/mongoose/issues/1474
-  if (organisation.wasNew) {
-    const createAdminRole = await Role.create({
-      title: 'Admin',
-      owner_id: organisation.owner,
-      organisation: organisation._id,
-      scopes: [ALL]
-    });
+  if (!organisation.wasNew) {
+    next();
+    return;
+  }
 
-    const createObserverRole = await Role.create({
-      title: 'Observer',
-      owner_id: organisation.owner,
-      organisation: organisation._id,
-      scopes: [
-        VIEW_PUBLIC_DASHBOARDS,
-        VIEW_PUBLIC_VISUALISATIONS,
-        VIEW_PUBLIC_JOURNEYS
-      ]
-    });
+  const createAdminRole = await Role.create({
+    title: 'Admin',
+    owner_id: organisation.owner,
+    organisation: organisation._id,
+    scopes: [ALL]
+  });
+
+  await Role.create({
+    title: 'Observer',
+    owner_id: organisation.owner,
+    organisation: organisation._id,
+    scopes: [
+      VIEW_PUBLIC_DASHBOARDS,
+      VIEW_PUBLIC_VISUALISATIONS,
+      VIEW_PUBLIC_JOURNEYS
+    ]
+  });
+
+  const owner = await User.findById(objectId(organisation.owner));
+
+  if (!owner) {
+    next();
+    return;
+  }
 
-    const owner = await User.findById(objectId(organisation.owner));
-    if (owner) {
-      // add this org to the creator's allowed orgs
-      const newRoles = [createAdminRole._id];
+  const newRoles = [createAdminRole._id];
 
-      const existingOrganisationSettings = owner.organisationSettings || [];
-      let updatedSettings = existingOrganisationSettings;
+  const existingOrganisationSettings = owner.organisationSettings || [];
+  let updatedSettings = existingOrganisationSettings;
 
-      // check if the user has settings for this org
-      const existingOS = find(existingOrganisationSettings, setting =>
-         setting.organisation.toString() === organisation._id.toString()
-      );
+  const existingOS = find(existingOrganisationSettings, setting =>
+    setting.organisation.toString() === organisation._id.toString()
+  );
+
+  if (existingOS) {
```

Comment:
In which case the existingOS variable will be true? As far as I understand the current orgSetting not exist in the owner yet. Maybe we only need to push the current orgSetting to the owner.orgSettings?

Reply:
Hmmmm thanks for flagging this, I'm actually not sure how we'd ever reach this code, I can only guess that it's there just in case the situation occurs somehow, perhaps while restoring an organisation that was deleted.

Reply:
I think this "if" statement never runs because as far as I can understand we have a bug inside "if". I will try to explain what I mean. Suppose the "updatedSettings" variable
```
  const existingOrganisationSettings = owner.organisationSettings || [];
  let updatedSettings = existingOrganisationSettings;
```
contains such an array
```
[organisationSettingObject1, organisationSettingObject2, organisationSettingObject3]
```
When we dive inside our "if" statement 
```
 if (existingOS) {
    const updatedOS = existingOS;

    updatedOS.roles = updatedOS.roles
      ? union(updatedOS.roles, newRoles)
      : updatedOS.roles = newRoles;

    updatedSettings = map(updatedSettings, (os) => {
      if (os.organisation.toString() === organisation._id.toString()) {
        return updatedOS;
      }
    });
```
we map over the "updatedSettings" variable. Because inside a map function we have "if" statement
```
      if (os.organisation.toString() === organisation._id.toString()) {
        return updatedOS;
      }
```
 the map function returns to us such an array 
```
[OrganisationSettingObject n, undefined, undefined, undefined...]. 
```
After that, we just put this array to the owner.orgSettings 
```
 owner.organisationSettings = updatedSettings;
```
save the owner and overwrite owner.orgSettings with "undefined" values in our database.

So I suppose testers noticed that while creating, updating, restoring organization data in owner overwritten

Reply:
I apologize if this is difficult to understand

Reply:
Yeah well spotted! No it wasn't difficult to understand, thanks for pointing that out and explaining it as clearly as you could. Since no one has reported this issue, I believe that is a decent enough indicator that this code is never reached, in which case let's remove this if block.

Reply:
Removing this if block will also get rid of the warning from the LGTM bot.
<img width="1237" alt="Screenshot 2020-02-17 at 08 22 00" src="https://user-images.githubusercontent.com/3045513/74635767-96fa8e80-515e-11ea-93cb-754144289843.png">

Reply:
Once you've removed this block @PrinceWaune please move this ticket to Ready for QA as it's already been reviewed by nearly everyone on the team at this point.

Reply:
Yeah, great!